### PR TITLE
Fix comment editing to use the right API.

### PR DESCRIPTION
**Pull request by rmmh**
It should be Issues.EditComment, not Repository.UpdateComment.

### CODE REVIEW

**Comment by rmmh** [comment]
Errors without this: 

    E1220 19:58:59.519577       1 github.go:1943] Error updating comment: PATCH https://api.github.com/repos/kubernetes/kubernetes/comments/268314989: 404 Not Found []

Woops!

**Comment by eparis** [comment]
While you are in here....

When we glog.Infof() the `body` we need to truncate that a bunch.
Before we actually send it to github we need to truncate it to < 65536

We do both "correctly" in `WriteComment()`

**Comment by eparis** [comment]
I also wonder if the truncating will mean that our comparator functions need some updating. But that's outside this particular area.

**Comment by rmmh** [comment]
OK, I made it smarter about how it handles too-long comments.

We should probably also truncate the failure reason to a few KB-- nobody wants to read stacktrace novels in a comment.

File: mungegithub/github/github.go [hunk 1]
```diff
@@ -1856,11 +1856,7 @@ func (obj *MungeObject) WriteComment(msg string) error {
 	config := obj.config
 	prNum := obj.Number()
 	config.analytics.CreateComment.Call(config, nil)
-	comment := msg
-	if len(comment) > 512 {
-		comment = comment[:512]
-	}
-	glog.Infof("Commenting in %d: %q", prNum, comment)
+	glog.Infof("Commenting in %d: %q", prNum, msg[:512])
```
Comment (eparis) [inline mungegithub/github/github.go]: Is this safe if len(msg) < 512?

**Review by eparis — CHANGES_REQUESTED** [review]

File: mungegithub/github/github.go (hunk 1 above)
Comment (rmmh) [inline mungegithub/github/github.go]: yes, it just becomes all of msg in that case

Comment (eparis) [inline mungegithub/github/github.go]: https://play.golang.org/p/Bl4MND4c4Q

Comment (rmmh) [inline mungegithub/github/github.go]: nope, I'm wrong :(

**Review by eparis — APPROVED** [review]

**Comment by eparis** [comment]
This looks safe. I'm not convinced the sync-er code is 'correct' in the face of large comments, but this is definitely 'better'

**Comment by eparis** [comment]
Who wrote these stupid bots? I said /lgtm AFTER the last code change (and before).  But it still removed my LGTM.

**Pull request merged**